### PR TITLE
SC-9090 - loading state and template refactoring

### PR DESCRIPTION
**Pull request by j-jmnz**
# Short Description
Implementation of a loading state and service-template refactoring

## Links to Ticket and related Pull-Requests
https://ticketsystem.hpi-schul-cloud.org/browse/SC-9090

## Checklist before merging

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] PO: Any deviation from requirements was agreed with Product-Owner / ticket author / support-team

> Notice: Please keep this Pull-Request as a Draft (or add WIP label), until it is ready to be reviewed

### CODE REVIEW

**Comment by j-jmnz** [comment]
@schul-cloud-bot update screenshots

**Comment by j-jmnz** [comment]
@schul-cloud-bot update screenshots

**Comment by j-jmnz** [comment]
@schul-cloud-bot update screenshots

File: src/pages/content/_id/index.vue [hunk 1]
```diff
@@ -23,11 +22,9 @@ export default {
 			: "plain";
 	},
 	async asyncData({ store, params }) {
```
Comment (pkleybolte) [inline src/pages/content/_id/index.vue:24]: Using `asyncData` is bad for two reasons:
1. It is nuxt specific (and in the long term we would like to get rid of it and use plain vue)
2. It blocks rendering the page until it is done. Better UX is rendering a page and showing a spinner.

Dispatch can be done in `mounted`. The returned object contains data from the store and should be mapped in `computed`.

If possible, please refactor. If it is too much effort now, please leave comments like this explanation in the code.

File: src/store/content.js [hunk 2]
```diff
@@ -102,9 +102,9 @@ export const actions = {
 			event.$emit("showModal@content", "errorModal");
 		}
 	},
-
-	async getResourceMetadata(context, id) {
-		return this.$axios.$get(`/edu-sharing/${id}`);
```
Comment (pkleybolte) [inline src/store/content.js:107]: I like that you noticed and fixed this :-)

**Review by pkleybolte — CHANGES_REQUESTED** [review]

**Review by pkleybolte — APPROVED** [review]

**Pull request merged**